agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -0,0 +1,124 @@
+package grpcclient
```

Comment:
nit taskresource/ doesn't seem like the right place for this

Reply:
since client is used by credentialspec, we felt this is the place to add gRPC client. if you have any other suggestion please advice